### DISCUSSION TITLE
Is it possible to have renovate migrate my shared preset json file?

### DISCUSSION THREAD

**Discussion by wessel-terpstra-vf** [renovatebot/renovate#28152] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure DevOps

### Please tell us more about your question or problem

I have a repostitory that contains a default.json with my renovate preset that is shared between different repositories on Azure DevOps. I have enabled the config migration setting and it migrates my renovate.json files successfully. However, it does not upgrade my default.json file that contains my shared preset. 

Is it possible to point renovate to my default.json shared preset file to make sure it will migrate that too?

**Comment by HonkingGoose** [comment]
The `configMigration` docs does not say if a `default.json` shared preset file will get config migrated or not:

> If enabled, Renovate raises a pull request when it needs to migrate the Renovate config file. Renovate only performs `configMigration` on `.json` and `.json5` files.

We should definitely explain this better in the docs. I don't know the correct answer, so I'll let a maintainer answer. 😄 

[Renovate docs for `configMigration` config option](https://docs.renovatebot.com/configuration-options/#configmigration)

**Comment by rarkins** [comment, marked as answer]
Renovate does not discover/migrate presets yet, but we should create a feature request for it if none exists

**Reply by HonkingGoose** [reply]
These issues seem a bit related:

- #22525
- #21456
- #11791

**Reply by rarkins** [reply]
Yes https://github.com/renovatebot/renovate/issues/22525 is it

**Comment by wessel-terpstra-vf** [comment]
Thanks for clarifying! Keep up the epic work :)

**Reply by HonkingGoose** [reply]
I'll re-open this discussion so it's easy to find for others who have the same problem. Once we actually support migrating shared presets, we can close the discussion for real. :smile:
